### ISSUE TITLE
mdadm.py module

### ISSUE THREAD

**Issue by gmolight** [saltstack/salt#23694]
running the mdadm state against minion in order to create a RAID device. While the device is successfully created the state returns the message below when trying to call the saved_config() method from mdadm module. 
```python
The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/salt/minion.py", line 1142, in _thread_return
        return_data = func(*args, **kwargs)
      File "/usr/lib/python2.7/dist-packages/salt/modules/mdadm.py", line 291, in save_config
        vol_d = dict([(line.split()[1], line) for line in scan])
    IndexError: list index out of range
```
I also tried the module directly and once again it works fine when creating a RAID device but fails when trying to save the config. Please advice

**Comment by jfindlay** [comment]
@gmolight, what is the salt version you are using, the output of `salt --versions-report` and `salt <minion> test.versions_report` as necessary?  Thanks.

**Comment by gmolight** [comment]
Sorry about not posting wiki the report.
OS: Ubuntu 14.04
master: 
                  Salt: 2015.5.0
                Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
 Debian source package: 2015.5.0+ds-1trusty1
minion:
                  Salt: 2015.5.0
                Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1

BTW, I did my own hack on it to get it working it seems that the regex that handles the return of the mdadm --list-scan is faulty when trying to parse the output. My solution is currently based on one record returned, however I was able to successfully removed the "buggy_ubuntu_tags" from the returned record, and successfully appended the record to the cfg_file. Please note that I did this mainly because I had to get these RAID objects deploy. I did not cleaned it up, did not know if it was good approach before I do a pull request. 

274     scan = __salt__['cmd.run']('mdadm --detail --scan', python_shell=False).split()
275     # Issue with mdadm and ubuntu
276     # REF: http://askubuntu.com/questions/209702/why-is-my-raid-dev-md1-showing-up-as-dev-md126-is-mdadm-conf-being-ignored
277     if __grains__['os'] == 'Ubuntu':
278         buggy_ubuntu_tags = ['name', 'metadata']
279         for attribute in scan[:]:
280             if attribute.startswith('name'):
281                 scan.remove(attribute)
282             elif attribute.startswith('metadata'):
283                 scan.remove(attribute)
284 
285         #for i, elem in enumerate(scan):
286         #    for bad_tag in buggy_ubuntu_tags:
287         #        pattern = r'\s{0}=\S+'.format(re.escape(bad_tag))
288         #        pattern = re.compile(pattern, flags=re.I)
289         #        scan[i] = re.sub(pattern, '', scan[i])
290 
291     if __grains__.get('os_family') == 'Debian':
292         cfg_file = '/etc/mdadm/mdadm.conf'
293     else:
294         cfg_file = '/etc/mdadm.conf'
295 
296     try:
297         __salt__['file.append'](cfg_file, '{entry}'.format(entry=' '.join(scan)))

**Comment by jfindlay** [comment]
@gmolight, no problem.  If you feel as if you're ready to submit a pull request, don't worry, we'll be sure to review it.  Your changes look mostly fine to me, but I'd probably remove the commented lines.  Also be sure to review https://docs.saltstack.com/en/latest/topics/development/contributing.html#which-salt-branch, thanks.

**Comment by gmolight** [comment]
Awesome thanks for the link. I'll review clean it up and give it a shot

**Comment by aryanet** [comment]
I have the same issue, and was just reading that same snippet of code. Why are we holding back on fixing this?

**Comment by aryanet** [comment]
I was just about to fix this, and I realized someone else beat me to it. The proper fix for this is already available on develop:;

https://github.com/saltstack/salt/commit/e44c04751b135d5028ee1a41d23d6924ef8d0460

**Comment by rallytime** [comment]
The commit referenced by @aryanet is for PR #24952, which should be present both on the 2015.8 branch (and all 2015.8.x releases) and the develop branch.

We can certainly back-port that fix to 2015.5. @gmolight Given that information would you consider this issue resolved?

**Comment by rallytime** [comment]
Since this is fixed in newer releases, and given the back-port PR above, I'm going to close this. Thanks everyone!
